src/views/Recorder/Recorder.tsx

```diff
@@ -176,6 +192,12 @@ export function Recorder() {
 
   useEffect(() => {
     return window.studio.browser.onBrowserClosed(async () => {
+      // if the user changed routes or closed the app during a recorder, the browser will be forced to close
+      // in this case, we don't need to save the recording again as it's already handled by handleConfirmNavigation
+      if (blocker.state === 'blocked' || isAppClosing) {
+        return
+      }
```

Comment:
This fixes a reproducible issue in `main` that saves two recording files when changing routes during a recording.

1. Start a recording
2. Change routes while the recording is in progress
3. Click "Stop recording"
4. Check that two recording files were saved


https://github.com/user-attachments/assets/03326e2f-723e-4c91-95a3-af9f8d7fcee3